### PR TITLE
LemonTree.Pipeline.Tools.ModelCheck

### PR DESCRIPTION
**Pull request by danielsiegl**
Introduced an Assembly that stores shared functionality and added a n…

### CODE REVIEW

**Comment by danielsiegl** [comment]
# ModelCheck results
 | Severity | Issue | Message |
 |Information|Model has 17 DIAGRAMIMAGEMAPS|This is perfect if you use it with WebEA and Prolaborate but makes merging/diffing harder|

**Comment by danielsiegl** [comment]
# ModelCheck results
| Severity | Issue | Message |
|----------|---------|---------|
|Information|Model has 17 DIAGRAMIMAGEMAPS|This is perfect if you use it with WebEA and Prolaborate but makes merging/diffing harder|
|Passed|No Baseline entries in the model||

**Comment by danielsiegl** [comment]
# LemonTree ModelCheck results
| Severity | Issue | Message |
|----------|---------|---------|
|Error|Model has 1 Baselines|Baselines are not helpful or required if you manage a model inside a VCS with LemonTree.|
|Information|Model has 17 DIAGRAMIMAGEMAPS|This is perfect if you use it with WebEA and Prolaborate but makes merging/diffing harder|
|Passed|No Audit entries in the model||
|Passed|User Security not enabled in the Model||

File: src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                result.Detail = $"Baselines are not helpful or required if you manage a model inside a VCS with LemonTree.";
+                result.Title = $"Model has {retVal} Baselines";
+
+            }
+
+
+            return result;
+        }
+
+        private static Issue checAudit(string model)
```
Comment (philippkalenda) [inline src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs:126]: typo in method name

File: src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                result.Detail = $"Audits are not helpful or required if you manage a model inside a VCS with LemonTree.";
+                result.Title = $"Model has {retVal} Audit Entires";
+
+            }
+
+
+            return result;
+        }
+
+        private static Issue checUserSecurity(string model)
```
Comment (philippkalenda) [inline src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs:148]: typo in method name

File: src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            Issue result = new Issue();
+            if (retVal == 0)
+            {
+                result.Level = IssueLevel.Passed;
+                result.Title = "User Security not enabled in the Model";
+            }
+            else
+            {
+                result.Level = IssueLevel.Warning;
+                result.Detail = $"User Security not enabled in the Model! Can cause higher complexity with LemonTree.";
```
Comment (philippkalenda) [inline src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs:161]: .... **is** enabled ...

File: src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+                result.Title = $"Model has {retVal} User Security Entires";
```
Comment (philippkalenda) [inline src/LemonTree.Pipeline.Tools.ModelCheck/Program.cs:162]: Entries

File: src/LemonTree.Pipeline.Tools.ModelCheck/CommandLineOptions/ModelCheckOptions.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using CommandLine;
+
+namespace LemonTree.Pipeline.Tools.ModelCheck.CommandLineOptions
+{
+    [Verb("ModelCheck", HelpText = "Checks if a Model is optimzed for LemonTree")]
```
Comment (philippkalenda) [inline src/LemonTree.Pipeline.Tools.ModelCheck/CommandLineOptions/ModelCheckOptions.cs:5]: optimized

**Review by philippkalenda — COMMENTED** [review]
just some typos

**Pull request merged**